### DISCUSSION TITLE
qrz and hamqth lookup very slow

### DISCUSSION THREAD

**Discussion by n5tti** [magicbug/Cloudlog#2574] [Q&A]
Good day,
I have added my login information for both qrz and hamqth for callbook lookups, however the lookup is very slow, most of the time taking 2 minutes to complete. 
While it is performing the lookup I am unable to save the QSO or navigate away from the page.
I have verified from the server that I can reach and successfully curl xmldata.qrz.com
during the lookup apache2 seems to be at 100% cpu on one core.  The server is configured with 8 cores and 8GB ram (so far it's using less than 1GB)
This appears to be the only issue. 
All other pages load normally, and the hardware interface is working correctly.
I am completely stumped ...

A screenshot of my debug screen below
![debug screenshot](https://github.com/magicbug/Cloudlog/assets/17775808/8666ca8b-c2c9-4889-a503-71cc7ea35c15)

Thanks for your help
Tom
N5TTI

**Comment by magicbug** [comment]
Seems excessive at the look up is with other calls, can you give us an idea of how many qsos are in your log?

**Reply by n5tti** [reply]
I have 3776 QSO's in my log. 
The majority of them were imported

**Comment by n5tti** [comment]
I was looking for logs, but when I navigate to application/logs/ the only file there is index.html
--
Edit - I did go to /var/log/apache2 and see the following error message.  The message appears 4 times and then immediately after the 4th error message the log page will update with the info from QRZ or hamqth -- 
`[Mon Oct 16 18:21:56.029118 2023] [php:error] [pid 1776] [client 192.168.1.69:61768] PHP Fatal error:  Maximum execution time of 30 seconds exceeded in /var/www/html/application/controllers/Lookup.php on line 84, referer: http://cloudlog/index.php/qso?manual=0`

**Comment by int2001** [comment]
maybe a ipv6 vs. ipv4 or Network issue?

**Reply by n5tti** [reply]
I don't think so since i was able to successfully curl xmldata.qrz.com 
Also the logs upload successfully when i upload them to qrz.com
I also temporarily disabled the firewall on the server to make sure that wasn't an issue.

**Reply by magicbug** [reply]
If you go into /application/config/config.php you can turn up the error logging level for Cloudlog that will then create error logs in application/log be interesting to see what its chucking out

**Reply by n5tti** [reply]
Here's the log after setting log level to 1 and entering a callsign (and then tabbing over)
`< ?php defined('BASEPATH') OR exit('No direct script access allowed'); ? >

ERROR - 2023-10-17 17:20:56 --> Severity: Warning --> Undefined array key "callsign" /var/www/html/application/controllers/Lookup.php 57
ERROR - 2023-10-17 17:20:56 --> Severity: Warning --> Undefined variable $uppercase_callsign /var/www/html/application/controllers/Lookup.php 66
ERROR - 2023-10-17 17:20:56 --> Severity: Warning --> Undefined variable $uppercase_callsign /var/www/html/application/controllers/Lookup.php 81
ERROR - 2023-10-17 17:21:26 --> Severity: Error --> Maximum execution time of 30 seconds exceeded /var/www/html/application/controllers/Lookup.php 84
ERROR - 2023-10-17 17:21:26 --> Severity: Warning --> Undefined array key "callsign" /var/www/html/application/controllers/Lookup.php 57
ERROR - 2023-10-17 17:21:26 --> Severity: Warning --> Undefined variable $uppercase_callsign /var/www/html/application/controllers/Lookup.php 66
ERROR - 2023-10-17 17:21:27 --> Severity: Warning --> Undefined variable $uppercase_callsign /var/www/html/application/controllers/Lookup.php 81
ERROR - 2023-10-17 17:21:56 --> Severity: Error --> Maximum execution time of 30 seconds exceeded /var/www/html/application/controllers/Lookup.php 84`

Here's the log after level 2 (debug) and entering a callsign (and then tabbing over)
`
DEBUG - 2023-10-17 17:24:17 --> cURL Class Initialized
DEBUG - 2023-10-17 17:24:19 --> cURL Class Initialized
DEBUG - 2023-10-17 17:24:20 --> cURL Class Initialized
DEBUG - 2023-10-17 17:24:23 --> cURL Class Initialized
DEBUG - 2023-10-17 17:24:24 --> cURL Class Initialized
DEBUG - 2023-10-17 17:24:26 --> cURL Class Initialized
DEBUG - 2023-10-17 17:24:29 --> cURL Class Initialized
DEBUG - 2023-10-17 17:24:29 --> cURL Class Initialized
ERROR - 2023-10-17 17:24:29 --> Severity: Warning --> Undefined array key "callsign" /var/www/html/application/controllers/Lookup.php 57
ERROR - 2023-10-17 17:24:29 --> Severity: Warning --> Undefined variable $uppercase_callsign /var/www/html/application/controllers/Lookup.php 66
ERROR - 2023-10-17 17:24:29 --> Severity: Warning --> Undefined variable $uppercase_callsign /var/www/html/application/controllers/Lookup.php 81
ERROR - 2023-10-17 17:24:59 --> Severity: Error --> Maximum execution time of 30 seconds exceeded /var/www/html/application/controllers/Lookup.php 84
DEBUG - 2023-10-17 17:24:59 --> cURL Class Initialized
ERROR - 2023-10-17 17:24:59 --> Severity: Warning --> Undefined array key "callsign" /var/www/html/application/controllers/Lookup.php 57
ERROR - 2023-10-17 17:24:59 --> Severity: Warning --> Undefined variable $uppercase_callsign /var/www/html/application/controllers/Lookup.php 66
ERROR - 2023-10-17 17:24:59 --> Severity: Warning --> Undefined variable $uppercase_callsign /var/www/html/application/controllers/Lookup.php 81
ERROR - 2023-10-17 17:25:29 --> Severity: Error --> Maximum execution time of 30 seconds exceeded /var/www/html/application/controllers/Lookup.php 84
DEBUG - 2023-10-17 17:25:29 --> cURL Class Initialized
DEBUG - 2023-10-17 17:25:29 --> cURL Class Initialized
DEBUG - 2023-10-17 17:25:30 --> Qrz class already loaded. Second attempt ignored.
DEBUG - 2023-10-17 17:25:30 --> Qra class already loaded. Second attempt ignored.
DEBUG - 2023-10-17 17:25:30 --> Qra class already loaded. Second attempt ignored.`

**Comment by n5tti** [comment]
I believe at this point i will blow it all away and re-install
that should correct everything
Thank you everyone for your suggestions
Tom
